core/src/mindustry/world/blocks/defense/turrets/LaserTurret.java

```diff
@@ -69,15 +69,15 @@ public void updateTile(){
             if(bullets.any()){
 
                 for(var entry : bullets){
-                    float
-                    bulletX = x + Angles.trnsx(rotation - 90, shootX + entry.x, shootY + entry.y),
-                    bulletY = y + Angles.trnsy(rotation - 90, shootX + entry.x, shootY + entry.y),
-                    angle = rotation + entry.rotation;
+                    float bulletX = x + Angles.trnsx(rotation - 90, shootX + entry.x, shootY + entry.y),
+                        bulletY = y + Angles.trnsy(rotation - 90, shootX + entry.x, shootY + entry.y),
+                        angle = rotation + entry.rotation;
 
                     entry.bullet.rotation(angle);
                     entry.bullet.set(bulletX, bulletY);
-                    entry.bullet.time = entry.bullet.type.lifetime * entry.bullet.type.optimalLifeFract;
-                    entry.bullet.keepAlive = true;
+                    float optLife = entry.bullet.type.lifetime * entry.bullet.type.optimalLifeFract;
```

Comment:
changing logic like this makes me very nervous

Reply:
it simply lets the bullet reach its optimal time on its own without snapping it to that poitn

Reply:
I have unchanged the logic.